### PR TITLE
[pytorch][build][sagemaker] update SMDDP wheel for pt 1.9

### PR DESCRIPTION
**Pull request by roywei**
### Description
We decided to still release SMDDP v1.2.0 instead of v1.2.1 for PT 1.9 since there is an issue with SM Profiler on PyTorch single node environment. 
Created PR to start the DLC pipeline, ~~pending performance tests on SMDDP side.~~ SMDDP locall and perf test passed, we are good to go.

### DLC image/dockerfile
PyTorch 1.9 Training GPU cu111 dockerfile

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -13,22 +13,22 @@ benchmark_mode = false
...
-eks_tests = true
-ec2_tests = true
+ecs_tests = false
+eks_tests = false
+ec2_tests = false
 
 # Off by default
-sagemaker_tests = false
-efa_tests = false
+sagemaker_tests = true
```
Comment (arjkesh) [inline dlc_developer_config.toml:31]: @roywei - if you consume the latest master, and set `sagemaker_remote_tests = "rc"` - you can also test RC tests

Comment (roywei) [reply dlc_developer_config.toml:31]: thanks! do you mean adding another line like this?
```
sagemaker_tests = true
sagemaker_remote_tests = rc
```

Comment (arjkesh) [reply dlc_developer_config.toml:31]: no, let me help resolve the conflicts and I'll show you

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -35,6 +35,6 @@ sagemaker_local_tests = false
 # "off" --> do not trigger sagemaker remote tests (default)
 # "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
 # "rc" --> run release_candidate_integration tests
-sagemaker_remote_tests = "off"
+sagemaker_remote_tests = "rc"
```
Comment (arjkesh) [inline dlc_developer_config.toml:38]: looks like rc tests pass - feel free to switch this to "standard" and we can debug any issues with standard suite of SM remote tests

**Review by saimidu — COMMENTED** [review]
Approved. Will revert toml changes and merge.

**Review by saimidu — APPROVED** [review]

**Review by mansimane — COMMENTED** [review]
@saimidu Did Sagemaker rubik and herring test run on this PR? I did not see them in for last few commits.

**Review by mansimane — APPROVED** [review]
LGTM. Discussed with @saimidu , few SMMP tests were run manually.

**Pull request merged**